pkg/varcontext/errorcollector.go

```diff
@@ -0,0 +1,54 @@
+// Copyright 2018 the Service Broker Project Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package varcontext
+
+import (
+	"errors"
+	"strings"
+)
+
+// ErrorCollector is a utility class that concatenates errors.
+// It is useful for builder-style objects that want to keep errors tracked
+// internally until a finalization call.
+type ErrorCollector struct {
+	errs []error
+}
+
+// HasErrors returns true if the collector has one or more errors.
+func (ec *ErrorCollector) HasErrors() bool {
+	return len(ec.errs) > 0
+}
+
+// Error gets a combined error message for all logged errors or nil if no errors
+// were logged.
+func (ec *ErrorCollector) Error() error {
+	if len(ec.errs) == 0 {
+		return nil
+	}
+
+	var out []string
+	for _, err := range ec.errs {
+		out = append(out, err.Error())
+	}
+
+	return errors.New(strings.Join(out, ", "))
```

Comment:
For improved readability during debugging, consider joining on '\n'.

Reply:
On second pass, I think serializing all the errors to a string might be hazardous. Go allows custom defined error types which could potentially contain more information than just a string. This limits the ability of upstream logic to benefit from such behavior. While such functionality might not be currently utilized, I suggest staying clear of decisions like this which limit your ability to utilize such language constructs completely.  Hashicorp has a package which might be of use to you in this situation: https://github.com/hashicorp/go-multierror